diff --git a/src/znc.cpp b/src/znc.cpp
index b33e860..4a02568 100644
--- a/src/znc.cpp
+++ b/src/znc.cpp
@@ -575,6 +575,7 @@ bool CZNC::WriteNewConfig(const CString& sConfigFile) {
 	VCString vsLines;
 
 	vsLines.push_back(MakeConfigHeader());
+	vsLines.push_back("PidFile = /run/znc/znc.pid");
 	vsLines.push_back("Version = " + CString(VERSION_STR));
 
 	m_sConfigFile = ExpandConfigPath(sConfigFile);
