diff --git a/src/ui/download.cc b/src/ui/download.cc
index 2e5e8c1..0cdcaf6 100644
--- a/src/ui/download.cc
+++ b/src/ui/download.cc
@@ -107,7 +107,7 @@ Download::create_menu() {
     [this] { activate_display_focus(DISPLAY_TRANSFER_LIST); },
     [this] { activate_display_menu(DISPLAY_TRANSFER_LIST); });
 
-  element->set_entry(0, false);
+  element->set_entry(1, false); // 'Info' active by default
 
   m_bindings['p'] = [element] { element->set_entry_trigger(0); };
   m_bindings['o'] = [element] { element->set_entry_trigger(1); };
@@ -242,7 +242,7 @@ Download::activate(display::Frame* frame, bool) {
   m_frame->frame(1)->initialize_window(m_windowDownloadStatus);
   m_windowDownloadStatus->set_active(true);
 
-  activate_display_menu(DISPLAY_PEER_LIST);
+  activate_display_menu(DISPLAY_INFO); // 'Info' active by default
 }
 
 void
