diff --git a/src/rpc/parse_commands.cc b/src/rpc/parse_commands.cc
index 2b9a9a2..c464717 100644
--- a/src/rpc/parse_commands.cc
+++ b/src/rpc/parse_commands.cc
@@ -234,6 +234,8 @@ parse_command_file(const std::string& path) {
     }
 
   } catch (torrent::input_error& e) {
+    if (!strcmp(e.what(), "import.return"))
+      return true;
     snprintf(buffer,
              2048,
              "Error in option file: %s:%u: %s",
