diff --git a/src/rpc/command_scheduler.cc b/src/rpc/command_scheduler.cc
index 5de26ca..f4ca947 100644
--- a/src/rpc/command_scheduler.cc
+++ b/src/rpc/command_scheduler.cc
@@ -30,15 +30,18 @@ CommandScheduler::insert(const std::string& key) {
   if (key.empty())
     throw torrent::input_error("Scheduler received an empty key.");
 
-  iterator itr = find(key);
-
-  if (itr == end())
-    itr = base_type::insert(end(), nullptr);
-  else
-    delete *itr;
+  CommandSchedulerItem* current = new CommandSchedulerItem(key);
+  current->slot() = std::bind(&CommandScheduler::call_item, this, current);
 
-  *itr           = new CommandSchedulerItem(key);
-  (*itr)->slot() = [this, item = *itr] { return call_item(item); };
+  iterator itr = find(key);
+  if (itr == end()) {
+    itr = base_type::insert(end(), current);
+  } else {
+    // swap in fully initialized command, and THEN delete the replaced one
+    CommandSchedulerItem* old = *itr;
+    *itr = current;
+    delete old;
+  }
 
   return itr;
 }
