diff --git a/src/core/manager.cc b/src/core/manager.cc
index 087efce..e333740 100644
--- a/src/core/manager.cc
+++ b/src/core/manager.cc
@@ -1,6 +1,7 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 // Copyright (C) 2005-2011, Jari Sundell <jaris@ifi.uio.no>
 
+#include <ctime>
 #include <cstdio>
 #include <cstring>
 #include <filesystem>
@@ -99,6 +100,27 @@ void
 Manager::push_log(const char* msg) {
   m_log_important->lock_and_push_log(msg, strlen(msg), 0);
   m_log_complete->lock_and_push_log(msg, strlen(msg), 0);
+  extern int log_messages_fd;
+  if (log_messages_fd >= 0) {
+    char buffer[30];
+    time_t t = std::time(0);
+    std::tm* now = std::localtime(&t);
+
+    snprintf(buffer,
+      sizeof(buffer),
+      "%04u-%02u-%02u %2d:%02d:%02d ",
+      1900 + now->tm_year,
+      now->tm_mon + 1,
+      now->tm_mday,
+      now->tm_hour,
+      now->tm_min,
+      now->tm_sec);
+
+    std::string line(buffer);
+    line += msg;
+    line += '\n';
+    ::write(log_messages_fd, line.c_str(), line.length());
+  }
 }
 
 Manager::Manager()
