diff --git a/include/rpc/object_storage.h b/include/rpc/object_storage.h
index 84c0fe2..e7b4c1b 100644
--- a/include/rpc/object_storage.h
+++ b/include/rpc/object_storage.h
@@ -173,6 +173,9 @@ public:
     list_push_back(torrent::raw_string::from_string(str), object);
   }
 
+  const torrent::Object& set_color_string(const torrent::raw_string& key,
+                                          const std::string& object);
+
   // Functions callers:
   torrent::Object call_function(const torrent::raw_string& key,
                                 target_type                target,
