diff --git a/include/ui/download_list.h b/include/ui/download_list.h
index ca09b22..98e974a 100644
--- a/include/ui/download_list.h
+++ b/include/ui/download_list.h
@@ -65,6 +65,9 @@ public:
   void disable() override;
 
   void activate_display(Display d);
+  ElementBase* display(Display d) {
+    return d < DISPLAY_MAX_SIZE ? m_uiArray[d] : 0;
+  }
 
   core::View* current_view();
   void        set_current_view(const std::string& name);
diff --git a/src/command_helpers.cc b/src/command_helpers.cc
index f14024f..7112593 100644
--- a/src/command_helpers.cc
+++ b/src/command_helpers.cc
@@ -24,6 +24,8 @@ initialize_command_peer();
 void
 initialize_command_local();
 void
+initialize_command_pyroscope();
+void
 initialize_command_logging();
 void
 initialize_command_network();
@@ -46,6 +48,7 @@ initialize_commands() {
   initialize_command_network();
   initialize_command_groups();
   initialize_command_local();
+  initialize_command_pyroscope();
   initialize_command_logging();
   initialize_command_ui();
   initialize_command_download();
