diff -Nur rtorrent_orig/src/rpc/object_storage.h rtorrent/src/rpc/object_storage.h
--- rtorrent_orig/src/rpc/object_storage.h	2015-09-09 17:17:48.763691769 +1200
+++ rtorrent/src/rpc/object_storage.h	2015-09-09 17:32:24.623678645 +1200
@@ -149,6 +149,8 @@
   void                   list_push_back(const torrent::raw_string& key, const torrent::Object& object);
   void                   list_push_back_str(const std::string& str, const torrent::Object& object) { list_push_back(torrent::raw_string::from_string(str), object); }
 
+  const torrent::Object& set_color_string(const torrent::raw_string& key, const std::string& object);
+
   // Functions callers:
   torrent::Object        call_function(const torrent::raw_string& key, target_type target, const torrent::Object& object);
   torrent::Object        call_function_str(const std::string& key, target_type target, const torrent::Object& object);
