diff -Nur rtorrent_orig/src/command_helpers.cc rtorrent/src/command_helpers.cc
--- rtorrent_orig/src/command_helpers.cc	2015-09-10 10:55:03.982741297 +1200
+++ rtorrent/src/command_helpers.cc	2015-09-14 14:44:54.517356425 +1200
@@ -49,6 +49,7 @@
 void initialize_command_ip();
 void initialize_command_peer();
 void initialize_command_local();
+void initialize_command_pyroscope();
 void initialize_command_logging();
 void initialize_command_network();
 void initialize_command_groups();
@@ -64,6 +65,7 @@
   initialize_command_network();
   initialize_command_groups();
   initialize_command_local();
+  initialize_command_pyroscope();
   initialize_command_logging();
   initialize_command_ui();
   initialize_command_download();
diff -Nur rtorrent_orig/src/Makefile.am rtorrent/src/Makefile.am
--- rtorrent_orig/src/Makefile.am	2015-09-10 10:55:03.982741297 +1200
+++ rtorrent/src/Makefile.am	2015-09-14 14:44:54.517356425 +1200
@@ -25,6 +25,7 @@
 	command_tracker.cc \
 	command_scheduler.cc \
 	command_ui.cc \
+	command_pyroscope.cc \
 	control.cc \
 	control.h \
 	globals.cc \
diff -Nur rtorrent_orig/src/ui/download_list.cc rtorrent/src/ui/download_list.cc
--- rtorrent_orig/src/ui/download_list.cc	2015-09-10 10:55:03.992741297 +1200
+++ rtorrent/src/ui/download_list.cc	2015-09-14 14:44:54.517356425 +1200
@@ -219,7 +219,7 @@
   // Set title.
   switch (displayType) {
   case DISPLAY_DOWNLOAD_LIST:
-    control->ui()->window_title()->set_title("rTorrent " VERSION "/" +
+    control->ui()->window_title()->set_title("rTorrent-PS " VERSION "/" +
                                              std::string(torrent::version()) + " - " +
                                              rpc::call_command_string("session.name"));
     break;
diff -Nur rtorrent_orig/src/ui/download_list.h rtorrent/src/ui/download_list.h
--- rtorrent_orig/src/ui/download_list.h	2015-09-10 10:55:03.992741297 +1200
+++ rtorrent/src/ui/download_list.h	2015-09-14 14:44:54.517356425 +1200
@@ -96,6 +96,7 @@
   void                disable();
 
   void                activate_display(Display d);
+  ElementBase*        display(Display d) { return d < DISPLAY_MAX_SIZE ? m_uiArray[d] : 0; }
 
   core::View*         current_view();
   void                set_current_view(const std::string& name);
