diff -Nur media_build_orig/v4l/scripts/make_config_compat.pl media_build/v4l/scripts/make_config_compat.pl
--- media_build_orig/v4l/scripts/make_config_compat.pl	2026-03-20 09:28:05.023998976 +1300
+++ media_build/v4l/scripts/make_config_compat.pl	2026-03-20 09:37:56.185991133 +1300
@@ -150,7 +150,7 @@
 	check_files_for_func("sysfs_emit", "NEED_SYSFS_EMIT", "include/linux/sysfs.h");
 	check_files_for_func("eth_hw_addr_set", "NEED_ETH_HW_ADDR_SET", "include/linux/etherdevice.h");
 	check_files_for_func("MODULE_IMPORT_NS", "NEED_MODULE_IMPORT_NS", "include/linux/module.h");
-	check_files_for_func("ALIGN_DOWN", "NEED_ALIGN_DOWN", "include/linux/align.h", "include/linux/kernel.h");
+	check_files_for_func("ALIGN_DOWN", "NEED_ALIGN_DOWN", "include/linux/align.h", "include/linux/kernel.h", "include/vdso/align.h");
 	check_files_for_func("bitmap_zalloc", "NEED_BITMAP_ZALLOC", "include/linux/bitmap.h");
 	check_files_for_func("firmware_request_nowarn", "NEED_FIRMWARE_REQUEST_NOWARN", "include/linux/firmware.h");
 	check_files_for_func("lockdep_assert_not_held", "NEED_LOCKDEP_ASSERT_NOT_HELD", "include/linux/lockdep.h");
