diff -Nur znc-push_orig/Makefile znc-push/Makefile
--- znc-push_orig/Makefile	2015-10-10 03:01:26.716184796 +0200
+++ znc-push/Makefile	2015-10-10 03:02:02.964766487 +0200
@@ -14,8 +14,8 @@
 		 znc-buildmod push.cpp
 
 install: push.so
-	mkdir -p $(HOME)/.znc/modules/
-	cp push.so $(HOME)/.znc/modules/push.so
+	mkdir -p $(DESTDIR)/modules/
+	cp push.so $(DESTDIR)/modules/push.so
 
 clean:
 	-rm -f push.so
