diff --git a/MANIFEST.in b/MANIFEST.in
index 5524869..7b49f90 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -9,7 +9,7 @@ include LICENSE
 
 recursive-include debian changelog
 recursive-include src/scripts *.sh *.sed
-recursive-include src/pyrocore/data/config rtorrent-*.rc
+recursive-include src/pyrocore/data/config *
 
 #recursive-include examples *.txt *.py
 #prune directory/pattern
diff --git a/pavement.py b/pavement.py
index bada087..8ad56db 100644
--- a/pavement.py
+++ b/pavement.py
@@ -79,7 +79,7 @@ project = Bunch(
     include_package_data = True,
     zip_safe = False,
     data_files = [
-        ("EGG-INFO", [
+        ("share/doc/pyrocore-9999", [
             "README.md", "LICENSE", "debian/changelog",
         ]),
     ],
