diff --git a/protonmail/protonmail.go b/protonmail/protonmail.go
index 8093fb3..685ef20 100644
--- a/protonmail/protonmail.go
+++ b/protonmail/protonmail.go
@@ -89,7 +89,7 @@ func (c *Client) newRequest(method, path string, body io.Reader) (*http.Request,
 		log.Printf(">> %v %v\n", req.Method, req.URL.Path)
 	}
 
-	req.Header.Set("X-Pm-Appversion", c.AppVersion)
+	req.Header.Set("X-Pm-Appversion", "Other")
 	req.Header.Set(headerAPIVersion, strconv.Itoa(Version))
 	c.setRequestAuthorization(req)
 	return req, nil
@@ -119,7 +119,7 @@ func (c *Client) newJSONRequest(method, path string, body interface{}) (*http.Re
 }
 
 func (c *Client) do(req *http.Request) (*http.Response, error) {
-	req.Header.Set("User-Agent", "Mozilla/5.0 (X11; Linux x86_64; rv:101.0) Gecko/20100101 Firefox/101.0")
+	req.Header.Set("User-Agent", "Ubuntu_20.04")
 
 	httpClient := c.HTTPClient
 	if httpClient == nil {
